feat: Revamp Search Bar UI [WIP]

senbo1 opened the pull request:
Fixes #5837

<!-- Add issue number above --> 

#### Describe the changes you have made in this PR -
Revamp's the UI of Navbar

### Todos
- [ ] Replace Select_tag with a custom select with stimulus controller to easily style the options dropdown and make it into a view component.

### Screenshots of the UI changes (If any) -
<!-- Do not add code diff here -->
<img width="1878" alt="Screenshot 2025-06-22 at 5 54 50 PM" src="https://github.com/user-attachments/assets/1946ae63-b14a-4f13-a211-d74367e55ebf" />


## Checklist before requesting a review
- [ ] I have added proper PR title and linked to the issue
- [ ] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests.




Note: Please check **Allow edits from maintainers.** if you would like us to assist in the PR.

senbo1 commented on app/components/search/search_bar_component.html.erb:
```diff
@@ -0,0 +1,26 @@
+    <%= text_field_tag :q, query, autocomplete: "off", class: "search-bar-input", data: { search_bar_target: "input" } %>
+    <button class="search-bar-button rounded-end" type="submit" name="button" value="Submit" aria-label="Search">
+      <svg width="20" height="21" viewBox="0 0 20 21" fill="none" xmlns="http://www.w3.org/2000/svg">
+        <path fill-rule="evenodd" clip-rule="evenodd" d="M8.74997 2.16666C7.62046 2.16676 6.50735 2.43697 5.50351 2.95474C4.49967 3.47252 3.63421 4.22285 2.97933 5.14313C2.32445 6.06341 1.89914 7.12695 1.73889 8.24503C1.57864 9.36311 1.68809 10.5033 2.05812 11.5705C2.42814 12.6377 3.04801 13.6009 3.866 14.3798C4.68399 15.1587 5.67639 15.7306 6.7604 16.048C7.8444 16.3654 8.98859 16.4189 10.0975 16 ...
```
no idea, need help on this

senbo1 commented on app/components/search/search_bar_component.html.erb:
```diff
@@ -0,0 +1,26 @@
+    </div>
+    <%= text_field_tag :q, query, autocomplete: "off", class: "search-bar-input", data: { search_bar_target: "input" } %>
+    <button class="search-bar-button rounded-end" type="submit" name="button" value="Submit" aria-label="Search">
+      <svg width="20" height="21" viewBox="0 0 20 21" fill="none" xmlns="http://www.w3.org/2000/svg">
+        <path fill-rule="evenodd" clip-rule="evenodd" d="M8.74997 2.16666C7.62046 2.16676 6.50735 2.43697 5.50351 2.95474C4.49967 3.47252 3.63421 4.22285 2.97933 5.14313C2.32445 6.06341 1.89914 7.12695 1.73889 8.24503C1.57864 9.36311 1.68809 10.5033 2.05812 11.5705C2.42814 12.6377 3.04801 13.6009 3.866 14.3798C4.68399 15.1587 5.67639 15.7306 6.7604 16.048C7.8444 16.3654 8.98859 16.4189 10.0975 16 ...
+      </svg>
```
added in new commit

senbo1 commented on app/components/search/search_bar_component.html.erb:
```diff
@@ -0,0 +1,26 @@
+    <div class="select-wrapper" data-search-bar-target="selectWrapper">
+      <button type="button" class="select-button" data-search-bar-target="selectButton" data-action="click->search-bar#toggleDropdown">
+        <span data-search-bar-target="selectedOption"><%= resource || resource_options.first %></span>
+        <svg class="select-arrow" viewBox="0 0 20 20" fill="currentColor">
+          <path fill-rule="evenodd" d="M5.293 7.293a1 1 0 011.414 0L10 10.586l3.293-3.293a1 1 0 111.414 1.414l-4 4a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414z" clip-rule="evenodd" />
+        </svg>
```
added in new commit

senbo1 commented on app/javascript/controllers/search_bar_controller.js:
```diff
@@ -11,11 +18,55 @@ export default class extends Controller {
+        selectedOption.classList.add('active');
+    }
+
+    handleOutsideClick(event) {
+        if (!this.selectWrapperTarget.contains(event.target)) {
+            this.closeDropdown();
+        }
+    }
```
should i add this as well @tachyons @vedant-jain03 ?

tachyons replied:
This seems to be  a nice to have

tachyons commented:
The dropdown is still not translated.

senbo1 commented on app/javascript/controllers/search_bar_controller.js:
```diff
@@ -2,20 +2,168 @@ import { Controller } from '@hotwired/stimulus';
+        this.setSelectedOption(event.currentTarget);
+    }
+
+    setSelectedOption(option) {
+        const { value } = option.dataset;
+
+        const optionLabelsMap = this.optionLabelsValue;
+        this.selectedOptionTarget.textContent = optionLabelsMap[value];
+        this.hiddenSelectTarget.value = value;
+
+        this.closeDropdown();
+
+        this.changePlaceholder();
+
+        this.updateActiveOption(option);
+    }
```
a value is always attached in a select option, and option won't be null as it's checked when passed by handleEnterKey and event handler always passes the event

senbo1 commented:
Added Select Options to internationalization
<img width="1597" alt="Screenshot 2025-06-28 at 7 44 56 AM" src="https://github.com/user-attachments/assets/5bfcc7cb-dd24-48e8-9aef-0d7e09d64bf8" />

Added Keyboard Navigation
https://github.com/user-attachments/assets/759e394e-1f65-43c1-8807-3801b0450276

Updated Lookbook Preview Layout (Reason - Search Bar Preview was not being overlapped with Header)
<img width="1597" alt="Screenshot 2025-06-28 at 7 50 30 AM" src="https://github.com/user-attachments/assets/22c79634-86e6-41ec-862a-f76d9c62fabb" />

senbo1 commented on app/assets/stylesheets/navbar.scss:
```diff
@@ -126,45 +126,136 @@
-  min-width: 150px;
-  padding: 5px;
+.search-bar-input {
+  border: 1px solid #E4E4E7;
```
adding in new commit

senbo1 commented:
all checks are passing, have made changes for the comments, please merge this @vedant-jain03

vedant-jain03 reviewed (approved):
🚀 LGTM!
